cmd/acr/purge.go

```diff
@@ -432,10 +434,12 @@ func getLastTagFromResponse(resultTags *acr.RepositoryTagsType) string {
 }
 
 // purgeDanglingManifests deletes all manifests that do not have any tags associated with them.
+// except the ones that are referenced by a multiarch manifest or that have subject.
 func purgeDanglingManifests(ctx context.Context, acrClient api.AcrCLIClientInterface, poolSize int, loginURL string, repoName string) (int, error) {
 	fmt.Printf("Deleting manifests for repository: %s\n", repoName)
 	// Contrary to getTagsToDelete, getManifestsToDelete gets all the Manifests at once, this was done because if there is a manifest that has no
-	// tag but is referenced by a multiarch manifest that has tags then it should not be deleted.
+	// tag but is referenced by a multiarch manifest that has tags then it should not be deleted. Or if a manifest has not tag, but it has subject,
+	// then it should not be deleted.
```

Comment:
@wju-MSFT I have a fundamental question here: What if a manifest has no tag but it has a subject where the subject has no tag? Should it be deleted?

---

Here's a case:
```mermaid
graph TD
  A -- subject --> B
  C-- manifests --> B
```
Should `A`, `B`, `C` be deleted?

---

Here's another case:
```mermaid
graph TD
  A -- subject --> B
  C-- manifests --> B
  Tag -.-> C
```
Should `A`, `B`, `C` be deleted?

Reply:
Basically, what will happen to the manifests that have no tags but have subjects to be deleted?

Reply:
discussed offline. will implement a solution to address the above corner cases in the next iteration.